File: src/main/java/gg/climb/lolobjects/game/ChampionImage.java
package gg.climb.lolobjects.game;

import org.immutables.value.Value;

import java.io.File;

/**
 * A class that represents a Champion's image. Construct an instance via
 * {@link ChampionImageBuilder}
 */
@Value.Style(visibility = Value.Style.ImplementationVisibility.PRIVATE,
             strictBuilder = true)
@Value.Immutable
public abstract class ChampionImage {

  /**
   * @return The file location of this champ's minimap image.
   */
  public abstract File getLocalFile();

  public abstract int getX();

  public abstract int getY();

  public abstract int getWidth();

  public abstract int getHeight();

  public abstract String getSprite();

  public abstract String getFull();

  public abstract String getGroup();
}

File: src/main/scala/gg/climb/lolobjects/game/ChampionImage.scala
package gg.climb.lolobjects.game

import java.io.File

/**
 * A class that represents a Champion's image.
 */
class ChampionImage(val file: File, val x: Int, val y: Int, val w: Int,
                    val h: Int, val sprite: String, val full: String, val group: String)

object ChampionImage{
  def apply(file: File, x: Int, y: Int,
            w: Int, h: Int, sprite: String,
            full: String, group: String) = new ChampionImage(file, x, y, w, h, sprite, full, group)
}
